packages/docusaurus/lib/server/load/config.js

```diff
@@ -32,6 +31,7 @@ const OPTIONAL_FIELDS = [
   'markdownPlugins',
   'plugins',
   'presets',
+  'themeConfig',
```

Comment:
Data that's meant to use in themes will all come under this key from now on.

Reply:
Shouldn't headerLinks fall in this themeConfig now ?